src/domain/platform/admin/space/AdminSpaceListPage/HostSelector.tsx

```diff
@@ -0,0 +1,176 @@
+import { FC, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import SearchIcon from '@mui/icons-material/Search';
+import { Autocomplete, FormHelperText, TextField } from '@mui/material';
+import { useOrganizationsListQuery, useUserSelectorQuery } from '../../../../../core/apollo/generated/apollo-hooks';
+import { CaptionSmall } from '../../../../../core/ui/typography';
+import FlexSpacer from '../../../../../core/ui/utils/FlexSpacer';
+import { ProfileChipView } from '../../../../community/contributor/ProfileChip/ProfileChipView';
+import { useField } from 'formik';
+
+const MAX_USERS_SHOWN = 10;
+
+interface HostFilterInput {
+  displayName?: string;
+}
+
+export interface Host {
+  type: 'user' | 'organization';
+  id: string;
+  profile: {
+    displayName: string;
+    location?: {
+      city: string;
+      country: string;
+    };
+    visual?: {
+      uri?: string;
+    };
+  };
+}
+
+interface HostSelectorProps {
+  name: string;
+  host?: Host;
+}
+
+export const mapUserOrOrganizationToHost = (
+  host:
+    | {
+        id: string;
+        profile: {
+          displayName: string;
+          location?: {
+            city: string;
+            country: string;
+          };
+          visual?: {
+            uri?: string;
+          };
+        };
+        __typename?: 'User' | 'Organization' | 'VirtualContributor';
+      }
+    | undefined
+): Host | undefined => {
+  if (!host || !host.__typename || (host.__typename !== 'User' && host.__typename !== 'Organization')) {
+    // We don't allouw VirtualContributors as hosts, at least for now
+    return undefined;
+  }
+
+  return {
+    id: host.id,
+    type: host.__typename === 'User' ? 'user' : 'organization',
```

Comment:
type: host.__typename.toLowerCase()?

Reply:
This would make possible the __typename to be any string, while the it should be only one of the 2 specified.